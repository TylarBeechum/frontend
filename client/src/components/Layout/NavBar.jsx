import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useStateValue } from 'react-conflux';
import { SecondaryButton } from '../../styles/themes';

import { useAuth0 } from '../../auth/authWrapper';
import { globalContext, LOGOUT } from '../../store/reducers/globalReducer';
import MenuLayer from './MenuLayer';
import stampdLogoWhite from '../../images/stampd_full_white.png';
import MenuButton from './MenuButton';

function NavBar({ history }) {
  const [{ onboarded }, globalDispatch] = useStateValue(globalContext);
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
  const [isShown, setShown] = useState(false);
  const [loading, setLoading] = useState(true);

  const toggleOpen = () => {
    setShown(!isShown);
    setLoading(false);
  };

  const handleLogout = () => {
    globalDispatch({
      type: LOGOUT,
      payload: false
    });
    localStorage.clear();
    setShown(false);
    setLoading(false);
    logout();
  };

  return (
    <NavContainter>
      <nav>
        <div className="left-section">
          {onboarded && (
            <MenuButton
              className="hamburger"
              setShown={setShown}
              isShown={isShown}
              onClose={toggleOpen}
              setLoading={setLoading}
              loading={loading}
            />
          )}
          <div className="logo">
            <Link to={onboarded ? '/dashboard' : '/'}>
              <img src={stampdLogoWhite} alt="Stampd logo" draggable="false" />
            </Link>
          </div>
        </div>
        <RightSection>
          <LinksSection>
            {!isAuthenticated && (
              <p>
                <Link to="/about">About Us</Link>
              </p>
            )}
            {!isAuthenticated && (
              <p>
                <Link to="/contact">Contact</Link>
              </p>
            )}
          </LinksSection>
          {!isAuthenticated ? (
            <div className="button__container">
              <NavBtn
                a11yTitle="Login"
                type="button"
                onClick={() => loginWithRedirect({})}
                label="Login"
              />
            </div>
          ) : (
            <div className="button__container">
              {/* <img src="" alt="avatar" /> */}
              <NavBtn
                a11yTitle="Logout"
                type="button"
                onClick={handleLogout}
                label="Logout"
              />
            </div>
          )}
        </RightSection>
      </nav>
      {isShown && (
        <MenuLayer
          loading={loading}
          setLoading={setLoading}
          toggleOpen={toggleOpen}
          isShown={isShown}
          setShown={setShown}
          history={history}
        />
      )}
    </NavContainter>
  );
}

NavBar.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const NavContainter = styled.div`
  display: flex;
  width: 100%;
  height: 70px;
  position: fixed;
  padding: 0 3%;
  justify-content: space-between;
  background-color: ${props => props.theme.global.colors.brand};
  align-items: center;
  top: 0;
  -webkit-box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  -moz-box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  z-index: 50;

  nav {
    margin: 0 auto;
    width: 100%;
    height: 70px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;

    .left-section {
      display: flex;
      align-items: center;

      .logo {
        width: 113px;
        height: auto;

        @media (max-width: 500px) {
          display: none;
        }

        a {
          display: flex;
          align-content: center;
        }
      }

      .hamburger {
        cursor: pointer;
      }
    }
  }
`;

const RightSection = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: space-between;
  width: 300px;

  p {
    color: white;
  }

  .button__container {
    right: 0;
    display: flex;
    justify-content: flex-end;
  }
`;

const NavBtn = styled(SecondaryButton)`
  border: 2px solid white /*${props => props.theme.global.colors['neutral-2']}*/;
  color: white /*${props => props.theme.global.colors['neutral-2']}*/;
  transition: background 0.3s, color 0.3s;

  :hover {
    color: ${({ theme }) => theme.global.colors.brand};
    background: ${({ theme }) => theme.global.colors.navbarHoverBg};
  }
`;

const LinksSection = styled.div`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin-right: 3%;
  p {
    margin-right: 7px;
    a {
      color: white;
      :hover {
        text-decoration: underline;
      }
    }
  }
`;

export default NavBar;
