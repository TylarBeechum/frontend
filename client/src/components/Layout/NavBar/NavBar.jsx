import React from 'react';
import { useStateValue } from 'react-conflux';
import styled from 'styled-components';

import { useAuth0 } from '../../Auth/authWrapper';
import { globalContext } from '../../../store/reducers/globalReducer';

function NavBar() {
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
  const [globalState] = useStateValue(globalContext);

  return (
    <NavContainter>
      <img src="" alt="logo" />
      <h1>{globalState.greeting}</h1>
      {!isAuthenticated ? (
        <div className="button-container">
          <button onClick={() => loginWithRedirect({})}>Log in</button>
        </div>
      ) : (
        <div className="button-container">
          <img src="" alt="avatar" />
          <button onClick={() => logout()}>Log out</button>
        </div>
      )}
    </NavContainter>
  );
}

const NavContainter = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid #333;
`;

export default NavBar;
