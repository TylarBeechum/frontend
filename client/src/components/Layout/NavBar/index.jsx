import React from 'react';
import { useStateValue } from 'react-conflux';
import styled from 'styled-components';
import { Menu } from 'grommet-icons';
import { BaseButton } from '../../../styles/themes';

import { useAuth0 } from '../../../auth/authWrapper';
import { globalContext } from '../../../store/reducers/globalReducer';

function NavBar(props) {
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
  const [globalState] = useStateValue(globalContext);

  return (
    <NavContainter>
      <Menu size="large" color="white" />
      {/* <img src="" alt="logo" /> */}
      <Logoh1>{globalState.greeting}</Logoh1>
      {!isAuthenticated ? (
        <div className="button-container">
          <LoginBtn
            {...props}
            a11yTitle="Login"
            type="button"
            onClick={() => loginWithRedirect({})}
            label="Login"
          />
        </div>
      ) : (
        <div className="button-container">
          <img src="" alt="avatar" />
          <BaseButton
            a11yTitle="Logout"
            type="button"
            onClick={() => logout()}
            label="Logout"
          />
        </div>
      )}
    </NavContainter>
  );
}

const NavContainter = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: #3aecfc;
  align-items: center;
  border-bottom: 2px solid #82fdff;
`;

const LoginBtn = styled(BaseButton)`
  border: 2px solid white /*${props => props.theme.global.colors['neutral-2']}*/;
  color: white /*${props => props.theme.global.colors['neutral-2']}*/;
`;

const Logoh1 = styled.h1`
  color: ${props => props.theme.global.colors['brand']};
`;

export default NavBar;

//${props => props.theme.global.colors['accent-3']}
