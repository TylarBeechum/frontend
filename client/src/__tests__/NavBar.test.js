/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import NavBar from '../components/Layout/NavBar';

describe('<NavBar />', () => {
  describe('should render the component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <NavBar history={{ location: { pathname: '/' } }} />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <NavBar history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should not render the navbar hamburger when a user is not signed in', () => {
    it('tests that the hamburger is not rendered when no user is signed in', () => {
      const { getByTestId } = renderWithRouterAndProviders(
        <NavBar history={{ location: { pathname: '/' } }} />
      );
      expect(getByTestId('hamburger')).toBeDefined();
    });
  });

  describe('should render the login button string to the screen successfully', () => {
    it('tests that the login button string is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <NavBar history={{ location: { pathname: '/' } }} />
      );
      const loginBtn = getByText(/logout/i);
      expect(loginBtn).toBeDefined();
    });
  });
});
