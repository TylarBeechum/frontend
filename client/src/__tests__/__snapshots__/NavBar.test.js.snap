// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-gzVnrw fUxDQo"
  >
    <nav>
      <div>
        <div
          className="sc-bZQynM kfMrJj"
          data-testid="hamburger"
        >
          <div
            className="hamburger-menu"
            onClick={[Function]}
          >
            <span />
            <span />
            <span />
          </div>
        </div>
        <div
          className="logo"
        >
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            <img
              alt="Stampd logo"
              draggable="false"
              src="stampd_full_white.svg"
            />
          </a>
        </div>
      </div>
      <div
        className="button__container"
      >
        <button
          aria-label="Logout"
          className="sc-bwzfXH sc-htoDjs hsGllO StyledButton-sc-323bzc-0 kHrfdt"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </nav>
  </div>
</div>
`;
